Security section [WIP] (#3)
hashbang/book: 2-Security/1-intro.md

## Security

Now that you (probably) know some stuff about the internet, it's valuable to
learn how you can protect yourself from people who want to attack you and your
services.

A long time ago, people used protocols across local networks to access files.
Assuming no one could intercept traffic, there was no reason to worry about

> whether or not someone could replace the files with a malicious verson.

Reviewer: ```suggestion
whether or not someone could replace the files with a malicious version.
```

Obviously, this is no longer the case, and more and more cases continue to rise
up about how people end up getting hacked and your personal data ends up thrown
out onto the internet.

> Software nowadays goes through a very long process to make it from the entity

Reviewer: This one reads a bit funny.

Maybe?
```suggestion
Software nowadays goes through a very long process to from the entity who made the software to your computer before you can use it.
```

Reviewer 2: remove first "to"

Author: Your suggested fix doesn't read right. I reworded it but "to from the entity" doesn't read right at all.

Reviewer: The rewording makes it much nicer. :+1:

who made the software, to your computer. In this process, it is often possible
for someone to slip up and make a mistake, leading to the entire system being
marked as invalid.

There are certain security measures that you should understand in order to make
sure that your data is as secure as possible. Throughout this chapter, we'll

> take a look at possible ways to attack, and prevent attacks against, a source

Reviewer 2: commas unnecessary

that you can't trust.

Fundamentally, trust is a problem that is hard to solve. How do you know that
your computer is really your computer, if you leave it out of sight for a few
minutes? How do you know that, when you go to Facebook, your ISP (internet
service provider) is not hijacking that domain? Even if you can look at a file
on one computer and copy it over to another computer, how do you know that your
flash drive didn't strip out a security measure?

---

In other chapters, we explain things in a format listing "the hard way" and
"the easy way". We will not be doing that in this chapter. Security is a very

> difficult field to get right, and off-putting for many people.

Reviewer 2: add "can be" to off-putting

> difficult field to get right, and off-putting for many people.

Reviewer: ```suggestion
difficult field to get right, and can be off-putting for many people.
```
